components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/util/JWTInboundUtil.java

```diff
@@ -0,0 +1,334 @@
+            }
+        }
+    }
+
+    public static void setBundleContext(BundleContext bundleContext) {
+
+        JWTInboundUtil.bundleContext = bundleContext;
+    }
+
+    public static void setRealmService(RealmService realmService) {
+
+        JWTInboundUtil.realmService = realmService;
+    }
+
+    public static void setConfigCtxService(ConfigurationContextService configCtxService) {
+
+        JWTInboundUtil.configCtxService = configCtxService;
+    }
+
+    public static void setHttpService(HttpService httpService) {
+
+        JWTInboundUtil.httpService = httpService;
+    }
+
+    public static void setResponseFactoryClass(String responseFactoryClass) {
+
+        JWTInboundUtil.responseFactoryClass = responseFactoryClass;
+    }
+
+    public static void setRequestFactoryClass(String requestFactoryClass) {
+
+        JWTInboundUtil.requestFactoryClass = requestFactoryClass;
+    }
+
+    public static void setRequestProcessorClass(String requestProcessorClass) {
+
+        JWTInboundUtil.requestProcessorClass = requestProcessorClass;
+    }
+
+    public static void setAuthConfigClass(String authConfigClass) {
+
+        JWTInboundUtil.authConfigClass = authConfigClass;
+    }
+
+    /**
+     * The method used to prevent CRLF_INJECTION_LOGS. It neutralizes the given input.
+     *
+     * @param input The input string to be neutralized.
+     * @return      neutralized output.
+     */
+    public static String neutralize(String input) {
+
+        if (StringUtils.isNotBlank(input)) {
+            return input.replaceAll("[\r\n]", "");
+        }
+        return null;
+    }
+
+    /**
+     * The method used to build the client side error page to be redirected to retry.do
+     *
+     * @return
+     */
+    public static JWTInboundResponse.JWTInboundResponseBuilder sendToRetryPage() {
+
+        JWTInboundResponse.JWTInboundResponseBuilder respBuilder = new JWTInboundResponse.JWTInboundResponseBuilder();
+        respBuilder.setEndpointUrl(ConfigurationFacade.getInstance().getAuthenticationEndpointRetryURL());
+        return respBuilder;
+    }
+
+    /**
+     * The method used to build the client side error page to be redirected to retry.do
+     *
+     * @param status    The status of the message defined in Resources.properties file of the authentication endpoint.
+     *                  Ex: misconfiguration.error
+     * @param statusMsg The message defined in Resources.properties file of the authentication endpoint.
+     *                  Ex: something.went.wrong.contact.admin
+     * @return
+     */
+    public static JWTInboundResponse.JWTInboundResponseBuilder sendToRetryPage(String status, String statusMsg) {
+
+        JWTInboundResponse.JWTInboundResponseBuilder respBuilder = new JWTInboundResponse.JWTInboundResponseBuilder();
+        respBuilder.setParameters(new HashMap<String, String>() {
+            {
+                put("status", status);
+                put("statusMsg", statusMsg);
```

Comment:
Move this constant to a constant file and use it from there?

Reply:
Fixed.